uniffi_bindgen/src/interface/types/mod.rs

```diff
@@ -62,6 +62,10 @@ pub enum Type {
     Optional(Box<Type>),
     Sequence(Box<Type>),
     Map(/* String, */ Box<Type>),
+    External {
+        name: String,
+        kind: ExternalTypeKind,
+    },
```

Comment:
This comment is going to assume a couple things for simplicity:
  - The code in my branch lands in some form or another
  - We adopt Ryan's proposal to rename `ViaFfi` to `FfiConverter`

Maybe this could be `ExternalFfiConverter` since we can support eternal types with Record/Enum/Object -- what's different here is the `FfiConverter` is located somewhere else.

Reply:
Thought about this some more and I think a) we may be able to standardize this some more and b) `ExternalFfiConverter` is an awkward name when you also need to append to it to name the kind

```
enum CustomType {
  # These might be enough to handle all of the scaffolding code
  ffi_converter_name: String,
  primitive: Box<Type>,

  # The bindings code is going to need more specialization
  bindings_kind: CustomTypeBindingsKind,
}

enum CustomTypeBindingsKind {
  StandardBinding(Box<Type>), // Handles the Guid -> String case
  ImportedFromOtherCrate(String /*path to udl file */),
  CustomTemplates(/* some story here about extension types */),
}
```

Reply:
I think your comment applies here to.  `ImportedFromOtherCrate` could point to a `Config` object or something similar.

Reply:
I'm struggling to find a good name of this one.  "CustomType" doesn't really communicate anything about what's going on.  Maybe `PrimitiveWrapper`?